src/cmds/qsub.c

```diff
@@ -2924,16 +2918,27 @@ process_opts(int argc, char **argv, int passet)
  *  rather than reading from a job script.
  *
  * @param[in]  argc         - argument count
- * @param[in]  argv         - pointer to array of argument variables
+ * @param[in]  argv         - pointer to original array of argument variables
+ * @param[in]  argv         - pointer to possibly-shuffled array argument variables
  * @param[out] script       - path of job script
  * @return     command_flag - indicates whether an executable was specified instead of a job script
  */
 static int
-process_special_args(int argc, char **argv, char *script)
+process_special_args(int argc, char **argv, char **argv_cpy, char *script)
 {
 	int command_flag = 0;
 	char *arg_list = NULL;
 	if (optind < argc) {
+		if (strcmp(argv[optind], argv_cpy[optind]) != 0) {
+			/* The arguments changed, the script and "--" must have been present.
+			 * getopt will move all non-options to the end of the array. In qsub's
+			 * case, it will only happen if both the "script" and "-- executable"
+			 * were present in the qsub command. This is unsupported usage and
+			 * should exit.
+			 */
+			print_usage();
+			exit_qsub(2);
+		}
```

Comment:
You passed in argv_cpy for just one check.  I'd put that check straight in main instead of passing in the copy.

Reply:
I put it in here because it looked out of place in main; I'll put it back.